ipa-core/src/net/server/mod.rs

```diff
@@ -467,9 +460,8 @@ impl<B, S: Service<Request<B>, Response = Response>> Service<Request<B>>
     }
 
     fn call(&mut self, mut req: Request<B>) -> Self::Future {
-        if let Some(header_value) = req.headers().get(&HTTP_HELPER_ID_HEADER) {
-            let id_result = ClientIdentity::try_from(header_value.clone())
-                .map_err(|e| Error::InvalidHeader(format!("{HTTP_HELPER_ID_HEADER}: {e}").into()));
+        if let Some(header_value) = req.headers().get(F::identity_header()) {
+            let id_result = ClientIdentity::<F::Identity>::try_from(header_value.clone());
```

Comment:
why do you need to clone it?

Reply:
Ok, I can use references as well (`TryFrom<&HeaderValue>`)